### DISCUSSION TITLE
unexpected result from auth()->user()->can('[permission-name]')

### DISCUSSION THREAD

**Discussion by mahmoudsabryksi** [spatie/laravel-permission#2473] [Q&A] [answered]
I have created a user with the role "Cashier" and this role contains a list of permissions when I try to check about certain permission it gives me "false" while I tried to list all of the user permissions I found the permission I checked for
![image](https://github.com/spatie/laravel-permission/assets/140489937/22cc8bc4-f267-4ef6-beec-7810168792fa)

**Comment by parallels999** [comment]
Refresh spatie cache, check guard_name

**Reply by parallels999** [reply]
instead of `$user->can('1') || $user->can('2')` you could use `canAny`, `$user->canAny('1', '2')`
also on blade `@canany`

**Reply by mahmoudsabryksi** [reply]
Thanks for your answer but can or canAny and @canany are not working either also I have Refresh spatie cache, check guard_name but still same output

**Reply by parallels999** [reply]
the advice was not to solve the problem, just for less code and better readability

**Comment by drbyte** [comment]
`can()` proxies to `hasPermissionTo()` but does a lookup based on the user's active guard and tries to find a permission for that guard and then checks whether the user has been granted that permission either directly or through a role.

So, like @parallels999 said, clearing the cache and inspecting your guards is a good place to start.

Further, you can replace `can()` with `hasPermissionTo()` and see whether the results change.

**Reply by mahmoudsabryksi** [reply]
thanks for your answer, can() and hasPermissionTo() return the same output false as long as adding a guard name to it

**Comment by mahmoudsabryksi** [comment]
If I check all permission given to users through direct permissions. "can and hasPermissionTo" works fine 
I think there is no check for all permissions given to users through role or direct permissions, I should do it by myself

**Reply by drbyte** [reply, marked as answer]
> I think there is no check for all permissions given to users through role or direct permissions,

You can see here that the `hasPermissionTo()` function checks both direct permissions and permissions via roles:
https://github.com/spatie/laravel-permission/blob/2cc68fdce917d7f7f6c37ec91157225afe9db902/src/Traits/HasPermissions.php#L225-L234

And calling `can` calls Laravel's `Gate`, which calls `checkPermissionTo()` which is a proxy to `hasPermissionTo()` (which bypasses any "permission not found" errors):

https://github.com/spatie/laravel-permission/blob/2cc68fdce917d7f7f6c37ec91157225afe9db902/src/PermissionRegistrar.php#L119-L132

https://github.com/spatie/laravel-permission/blob/2cc68fdce917d7f7f6c37ec91157225afe9db902/src/Traits/HasPermissions.php#L275-L282

**Reply by mahmoudsabryksi** [reply]
thanks, Drbyte for your help, By the way, my issue was on overriding role and permission models in wrong way I have fixed the overriding issue
